src/main/java/nitezh/ministock/activities/ChartActivity.java

```diff
@@ -24,18 +38,92 @@
 public class ChartActivity extends Activity {
     // Public variables
     public static int mAppWidgetId = 0;
+    private final String alphavantagekey = "ZKD8M6L9CEQAK89H";
 
     @Override
     public void onCreate(Bundle savedInstanceState) {
         URL url = null;
         Bitmap bmImg = null;
+        int interval = GlobalWidgetData.getInterval();
         int position = getIntent().getIntExtra(WidgetProviderBase.ROW_POSITION, 0);
+        //MyData.setInterval(2); /*= getIntent().getIntExtra(WidgetProviderBase.INTERVAL, 2);*/
         super.onCreate(savedInstanceState);
         setContentView(R.layout.bonobo_chart_layout);
-        Spanned html = Html.fromHtml("Sample Chart View (Row " + String.valueOf(position) + ") <br /><br />");
+        String symbol = GlobalWidgetData.getList().get(position).getSymbol();
+        Spanned html = Html.fromHtml("Graph of " + symbol + " <br /><br />");
         TextView text = (TextView) findViewById(R.id.chart_text);
         text.setText(html);
-        new ImageSnatcher((ImageView) findViewById(R.id.chart_img)).execute("https://www.codeproject.com/KB/graphics/zedgraph/example_1.png");
+        String intervalStr = intervalSwitcher(interval);
+        Log.i("urltext", "https://www.alphavantage.co/query?function=TIME_SERIES_" + intervalStr + "_ADJUSTED&symbol="
+                + symbol + "&apikey=" + alphavantagekey);
+
+
+        List<String> prices = GlobalWidgetData.getValues("https://www.alphavantage.co/query?function=TIME_SERIES_" + intervalStr + "_ADJUSTED&symbol="
+                + symbol + "&apikey=" + alphavantagekey, interval);
+
+        new ImageSnatcher((ImageView) findViewById(R.id.chart_img)).execute(GlobalWidgetData.constructImageUrl(prices));
+
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        int interval = GlobalWidgetData.getInterval();
+        int position = getIntent().getIntExtra(WidgetProviderBase.ROW_POSITION, 0);
+        Button but7day = (Button) findViewById(R.id.but_7day);
```

Comment:
I think "btn" is a better and more universal acronym than "but" for buttons

Reply:
Note: Every request refers to the last line shown (73 in this case)

Reply:
fixed